lib/easypost/http_client.rb

```diff
@@ -27,11 +27,16 @@ def request(
     request.body = JSON.dump(EasyPost::InternalUtilities.objects_to_ids(body)) if body
 
     # Execute the request, return the response.
-    Net::HTTP.start(
-      uri.host,
-      uri.port, use_ssl: true, read_timeout: @config[:read_timeout], open_timeout: @config[:open_timeout],
-    ) do |http|
-      http.request(request)
+
+    begin
+      Net::HTTP.start(
+        uri.host,
+        uri.port, use_ssl: true, read_timeout: @config[:read_timeout], open_timeout: @config[:open_timeout],
+      ) do |http|
+        http.request(request)
+      end
+    rescue Net::ReadTimeout, Net::OpenTimeout => e
+      raise EasyPost::Exceptions::TimeoutError.new(e.message)
```

Comment:
question: don't we need to catch other HTTP errors here as the last line of defence? something that isn't caught by a status code for instance that raises an error should fall back to the `HttpError` right?

Reply:
Resolved